### ISSUE TITLE
Rubocop-Minitest assert_offense breaks based off #11665 until new release

### ISSUE THREAD

**Issue by broksonic21** [rubocop/rubocop#11670]
## Expected behavior

Rubocop 1.48.0 should drop in and work as prior versions (1.45.1 if you were waiting on the perf fix from #11657)

## Actual behavior

rubocop_minitest fails on the following call stack when you are testing a custom cop:

```
NoMethodError: NoMethodError: undefined method `for_cop' for nil:NilClass
    /REDACTED_PATH/bundle/gems/rubocop-1.48.0/lib/rubocop/cop/registry.rb:202:in `enabled?'
    /REDACTED_PATH bundle/gems/rubocop-1.48.0/lib/rubocop/cop/team.rb:164:in `block in roundup_relevant_cops'
    /REDACTED_PATH/bundle/gems/rubocop-1.48.0/lib/rubocop/cop/team.rb:161:in `select'
    /REDACTED_PATH/bundle/gems/rubocop-1.48.0/lib/rubocop/cop/team.rb:161:in `roundup_relevant_cops'
    /REDACTED_PATH/bundle/gems/rubocop-1.48.0/lib/rubocop/cop/team.rb:88:in `investigate'
    /REDACTED_PATH/bundle/gems/rubocop-minitest-0.28.0/lib/rubocop/minitest/assert_offense.rb:128:in `_investigate'
    /REDACTED_PATH/bundle/gems/rubocop-minitest-0.28.0/lib/rubocop/minitest/assert_offense.rb:119:in `assert_offense'
```

## Steps to reproduce the problem

Run with minitest 4.28.0 and use assert_offense.

Note: It is _**already fixed**_, just not released https://github.com/rubocop/rubocop-minitest/commit/fcf0a8f33739f88b8ac4aaca0e4d47977e2fc76d

So, basically, rubocop-minitest 0.28.0 and rubocop 1.48.0 (or really 1.46.0+ I believe) are not compatible until a new version of rubocop-minitest is cut.

## RuboCop version

```
1.48.0 (using Parser 3.2.1.0, rubocop-ast 1.27.0, running on ruby 2.7.7) [arm64-darwin21]
  - rubocop-minitest 0.28.0
```

**Comment by broksonic21** [comment]
fyi  @fatkodima @koic @bbatsov  , only fallout I've noticed from the perf change yesterday. Thanks again for quick turnaround!

**Comment by koic** [comment]
RuboCop Minitest 0.29 has been released:
https://github.com/rubocop/rubocop-minitest/releases/tag/v0.29.0

Please upgrade RuboCop Minitest to the latest version. Thank you.

**Comment by broksonic21** [comment]
Thanks @koic , confirming that fixed the issue for us!
